docs/concepts/models/model_kinds.md

```diff
@@ -334,8 +334,10 @@ MODEL (
   name db.employees,
   kind INCREMENTAL_BY_UNIQUE_KEY (
     unique_key name,
-    when_matched WHEN MATCHED AND source.value IS NULL THEN UPDATE SET target.salary = COALESCE(source.salary, target.salary),
-    WHEN MATCHED THEN UPDATE SET target.title = COALESCE(source.title, target.title)
+    when_matched (
```

Comment:
do you need the parenthesis?

Reply:
Yeah, I think we do. For example:

```sql
MODEL (
  name db.employees,
  kind INCREMENTAL_BY_UNIQUE_KEY (
    when_matched WHEN MATCHED THEN UPDATE SET target.salary = COALESCE(source.salary, target.salary),
    unique_key name
  )
);

SELECT 'name' AS name, 1 AS salary;
```

This fails to parse because the comma is consumed while parsing the `UPDATE SET` [clause](https://github.com/tobymao/sqlglot/blob/main/sqlglot/parser.py#L7047-L7051), so you end up consuming `unique_key` and `name` looks like a stray, unknown property afterwards:

```
[Property(
  this=when_matched,
  value=[
    When(
      matched=True,
      then=Update(
        expressions=[
          EQ(
            this=Column(
              this=Identifier(this=salary, quoted=False),
              table=Identifier(this=target, quoted=False)),
            expression=Coalesce(
              this=Column(
                this=Identifier(this=salary, quoted=False),
                table=Identifier(this=source, quoted=False)),
              expressions=[
                Column(
                  this=Identifier(this=salary, quoted=False),
                  table=Identifier(this=target, quoted=False))])),
          Column(
            this=Identifier(this=unique_key, quoted=False))]))])] <~~~
```

I think models that use `when_matched` must've been working by coincidence until now, e.g. by having it placed at the end of the model kind's attributes.

The parentheses solve this issue by making the delimiters around the `WHEN MATCHED` clauses explicit.